subscriptions/programs/subscriptions/src/contexts/withdraw.rs

```diff
@@ -0,0 +1,66 @@
+use {
+    crate::{error::ErrorCode, state::*},
+    anchor_lang::prelude::*,
+    anchor_spl::token::{self, Mint, Token, TokenAccount, Transfer},
+};
+
+#[derive(Accounts)]
+pub struct Withdraw<'info> {
+    #[account(mut)]
+    pub payer: Signer<'info>,
+    #[account(
+        mut,
+        token::mint = mint,
+        token::authority = payer,
+    )]
+    pub payer_token_account: Account<'info, TokenAccount>,
+
+    #[account(mut, address = Subscription::pda(subscription.owner.key(),subscription.subscription_id.clone()).0)]
+    pub subscription: Account<'info, Subscription>,
+    #[account(
+        mut,
+        token::mint = mint,
+        token::authority = subscription,
+        address = Subscription::bank_pda(subscription.key(),subscription.owner.key()).0
+    )]
+    pub subscription_bank: Account<'info, TokenAccount>,
+
+    #[account(address=subscription.mint)]
+    pub mint: Account<'info, Mint>,
+    pub token_program: Program<'info, Token>,
+}
+
+impl<'info> Withdraw<'_> {
+    pub fn process(&mut self) -> Result<()> {
+        let Self {
+            payer,
+            subscription,
+            token_program,
+            payer_token_account,
+            subscription_bank,
+            ..
+        } = self;
+
+        require!(subscription.owner == payer.key(), ErrorCode::NotOwner);
```

Comment:
This should be put in the account validation. Rename `payer` to `owner` and put a `has_one = owner` constraint on the `subscription` account.